Fix get profile after logging in (#27)
vtex-apps/login: docs/README.md

|        `identifierPlaceholder`        | `string`  |                                                              Text that will display as a placeholder for the extension input.                                                               |  `undefined`  |
|       `invalidIdentifierError`        | `string`  |                                                                         Error message for invalid user identifier.                                                                          |  `undefined`  |
|         `termsAndConditions`          | `string`  |                                                       Text that will display below the login options regarding terms and conditions.                                                        |  `undefined`  |
|         `disabledEmailInputs`         | `boolean` |                                                     Determines whether user email editing should be allowed (`true`) or not (`false`).                                                      |    `false`    |

### Advanced configuration

By default, the email/password login asks two inputs from users:

- Email (_user identifier_)
- Password

> You can **set up the User Identifier Extension to allow other identifiers to be used in the Login form**, as long as they can be resolved to the user email in the background.

Reviewer: ```suggestion
You can set up the User Identifier Extension to allow other identifiers to be used in the Login form as long as they can be resolved to the user email in the background.
```

For example: If your account stores the national identity document of each user, the email/password login could be changed to ask for the following two inputs:

- National identity document (_user identifier_)
- Password

In order to display an identifier other than the user email, you have to create an **extension app** of your own to run in the background alongside the Login component.

The new extension app should be set up to receive the custom user identifier from the interface, such as the national identity document, and link it to the right user using store data. After that, the app should fetch the user email and return it to the Login app, which will behave as if the user had entered their email.

Below, you can find the instructions for running the User Identifier Extension in your store:

> > ⚠️ _Only enterprise clients can use this extension. You can check your account plan in your VTEX contract._

Reviewer: ```suggestion
> ⚠️ Only enterprise clients can use this extension. You can check your account plan in your VTEX contract.
```

},
```

2. [Develop a new component](https://vtex.io/docs/getting-started/desenvolva-componentes-usando-vtex-io-e-react/1) to replace the `email` input in the login form. This can be anything you want in its place, such as a file uploader or a simple text input which gets the national identity document.

> > ℹ️ _When developing the component, remember that it must know how to convert the custom identifier to an email. This should be done by adding a callback function that returns an email within the Login app in the `react/{{ComponentName}}.js` file. For example:_

Reviewer: ```suggestion
When developing the component, remember that it must know how to convert the custom identifier to an email. This should be done by adding a callback function that returns an email within the Login app in the `react/{{ComponentName}}.js` file. For example:
```

  const onChangeInput = useCallback(
    (e) => setInputValue(e.target.value),
    [setInputValue]
  );

  // This callback function (onSubmit) should return the resolved email. In the example below, it adds `@email.com` to the current
  // value in the Input component (e.g.: "john" would be resolved to "john@email.com").
  const onSubmit = useCallback(async () => {
    const email = `${inputValue}@email.com`;
    return email;
  }, [inputValue]);

  // This code adds the async callback function you defined in the login app when this component mounts.
  useEffect(() => {
    registerSubmitter(onSubmit);
  }, [registerSubmitter, onSubmit]);

  // The component calls "renderInput" and renders its result.
  return renderInput({
    value: inputValue,
    onChange: onChangeInput,
    placeholder: identifierPlaceholder,
  });
};

export default ComponentName;
```

3. In the `store/interfaces.json` file, create a new [interface](https://developers.vtex.com/docs/guides/vtex-io-documentation-interface) for `user-identifier`, declaring the component you created in the previous step:

```json
"user-identifier.{{InterfaceName}}": {
  "component": "{{ComponentName}}"
},
```

> > ℹ️ _The interface name, which is represented in the example above as `{{InterfaceName}}` can be anything your want._ ℹ️ _In the example above, the component created in step 2 is represented by `{{ComponentName}}`._

Reviewer: ```suggestion
> ℹ️ The interface name, which is represented in the example above as `{{InterfaceName}}` can be anything your want. In the example above, the component created in step 2 is represented by `{{ComponentName}}`.
```

4. In the `store/plugins.json` file, add the new interface (`user-identifier`) for the Login app:

```json
"login > user-identifier": "user-identifier.{{InterfaceName}}",
"login-content > user-identifier": "user-identifier.{{InterfaceName}}",
```

> > ℹ️ _If the `store/plugins.json` file is not available, you should create it._

Reviewer: ```suggestion
> ℹ️ If the `store/plugins.json` file is not available, you should create it.
```

> > ⚠️ _Since the Login app behaves as if users were typing their emails, error messages like `The email is invalid` may appear even if the component is requesting other identifiers from users. This message, and others, can be edited in the Admin CMS._

Reviewer: ```suggestion
> ⚠️ Since the Login app behaves as if users were typing their emails, error messages like `The email is invalid` may appear even if the component is requesting other identifiers from users. This message, and others, can be edited via the Store Editor in the Admin.
```
